app/src/main/java/com/dindin/musicapp/ui/main/MainActivity.kt

```diff
@@ -0,0 +1,169 @@
+            SongRepository.getInstance(SongLocalDataSource.getInstance(this))
+        )
+        if (checkPermission()) {
+            init()
+        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            requestPermissions(permissions, 0)
+        }
+        registerReceiver(serviceReceiver, IntentFilter(getString(R.string.action_receiver_service)))
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        unregisterReceiver(serviceReceiver)
+    }
+
+    override fun setSongs(songList: List<Song>) {
+        songs = songList as ArrayList<Song>
+    }
+
+    override fun updateRecyclerSong(songList: List<Song>) {
+        binding.recyclerSong.apply {
+            layoutManager = LinearLayoutManager(
+                this@MainActivity,
+                RecyclerView.VERTICAL,
+                false
+            )
+            adapter = SongAdapter(songList) {
+                startService(it)
+            }
+        }
+    }
+
+    override fun showMessage(message: String) {
+        binding.textMessage.apply {
+            text = message
+            visibility = View.VISIBLE
+        }
+    }
+
+    override fun hideMessage() {
+        binding.textMessage.visibility = View.GONE
+    }
+
+    private fun init() {
+        checkPermission()
+        presenter?.getSongsFromLocal()
+    }
+
+    private fun handleLayoutMusic(index: Int, playing: Boolean, action: Int) {
+        when (action) {
+            MyService.ACTION_START, MyService.ACTION_NEXT -> {
+                binding.groupPlayMusic.visibility = View.VISIBLE
+                showInfoSong(index)
+                setStatusButtonPlay(playing)
+            }
+            MyService.ACTION_PAUSE -> {
+                setStatusButtonPlay(playing)
+            }
+            MyService.ACTION_RESUME -> {
+                setStatusButtonPlay(playing)
+            }
+            MyService.ACTION_EXIT -> {
+                binding.groupPlayMusic.visibility = View.GONE
+                stopService()
+            }
+        }
+    }
+
+    private fun showInfoSong(index: Int) {
+        binding.apply {
+            textSongName.text = songs[index].title
+            textSongArtist.text = songs[index].artist
+            buttonNext.setOnClickListener {
+                sendToService(MyService.ACTION_NEXT)
+            }
+        }
+    }
+
+    private fun setStatusButtonPlay(playing: Boolean) {
+        binding.buttonPlay.apply {
+            if (playing) {
+                setImageResource(R.drawable.ic_pause)
+                setOnClickListener {
+                    sendToService(MyService.ACTION_PAUSE)
+                }
+            } else {
+                setImageResource(R.drawable.ic_play)
+                setOnClickListener {
+                    sendToService(MyService.ACTION_RESUME)
+                }
+            }
+        }
+    }
+
+    private fun sendToService(action: Int) {
+        val intentService = Intent(this, MyService::class.java)
```

Comment:
```suggestion
        val intentService = MyService.getIntent(this)
```
xem lại những chỗ khác nữa nhé

Reply:
có thể dùng với tất cả class hay chỉ với mỗi service hả anh

Reply:
được hết e nhé

Reply:
tym tym